src/target/linux.rs

```diff
@@ -306,6 +306,22 @@ impl LinuxTarget {
         )
     }
 
+    /// unmap the memory.
+    pub unsafe fn munmap(
+        &self,
+        addr: *mut libc::c_void,
+        length: libc::size_t,
+    ) -> CrabResult<()> {
+        if libc::munmap(
```

Comment:
This would unmap the memory inside the debugger process, not the debugged process. You need to use `self.syscall(libc::SYS_unmap as _, addr as _, length as _)`

Reply:
uh...i guess you are right since i dont know much about this 


but, what about the additional arguments? i initially wanted to call `self.syscall`, but could not figure out what should i pass to the additional arguments. and i was afraid to pass any value, since the code for self.syscall seems to be modifying registers.

Reply:
You can pass in 0. `self.syscall` restores all registers to their original value after performing the syscall: https://github.com/headcrab-rs/headcrab/blob/8e9424a8a9d650a6a0242b0cbbb5f8e7388f9037/src/target/linux.rs#L283

Reply:
ah yes, thanks for the reassurance and directions!